src/bluesky/callbacks/_mpl_image_cross_section/cross_section.py

```diff
@@ -0,0 +1,469 @@
+from matplotlib.widgets import Cursor
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+from bluesky.callbacks._mpl_image_cross_section.interpolation import InterpolationEnum
+from bluesky.callbacks._mpl_image_cross_section.limit_factories import fullrange_limit_factory
+
+logger = logging.getLogger(__name__)
+
+
+def auto_redraw(func):
+    """
+    decorator to automatically redraw the figure after a function call
+    """
+
+    def inner(self, *args, **kwargs):
+        if self._fig.canvas is None:
+            return
+        force_redraw = kwargs.pop("force_redraw", None)
+        if force_redraw is None:
+            force_redraw = self._auto_redraw
+
+        ret = func(self, *args, **kwargs)
+
+        if force_redraw:
+            self._update_artists()
+            self._draw()
+
+        return ret
+
+    inner.__name__ = func.__name__
+    inner.__doc__ = func.__doc__
+
+    return inner
+
+
+class CrossSection:
+    """
+    Class to manage the axes, artists and properties associated with
+    showing a 2D image, a cross-hair cursor and two parasite axes which
+    provide horizontal and vertical cross sections of image.
+
+    You will likely need to call `CrossSection.init_artists(init_image)` after
+    creating this object.
+
+    Parameters
+    ----------
+
+    fig : matplotlib.figure.Figure
+        The figure object to build the class on, will clear
+        current contents
+
+    cmap : str,  colormap, or None
+        color map to use.  Defaults to gray
+
+    norm : Normalize or None
+       Normalization function to use
+
+    limit_func : callable, optional
+        function that takes in the image and returns clim values
+    auto_redraw : bool, optional
+    interpolation : str, optional
+        Interpolation method to use. List of valid options can be found in
+        CrossSection2DView.interpolation
+    aspect : str, optional
+        Aspect passed into imshow, defaults to equal
+
+    Properties
+    ----------
+    interpolation : str
+        The stringly-typed pixel interpolation. See _INTERPOLATION attribute
+        of this cross_section_2d module
+    cmap : str
+        The colormap to use for rendering the image
+
+
+    """
+
+    def __init__(
+        self,
+        figure: Figure,
+        colormap: Optional[Colormap] = None,
+        norm=None,
+        limit_func: Optional[Callable[[Optional[Any]], Callable[[Any], tuple]]] = None,
+        auto_redraw=True,
+        interpolation: InterpolationEnum = InterpolationEnum.NONE,
+        aspect="equal",
+    ):
+        self._cursor_position_cbs: List[Any] = []
+        self._interpolation = interpolation
+        # used to determine if setting properties should force a re-draw
+        self._auto_redraw = auto_redraw
+        # clean defaults
+        if limit_func is None:
+            # note - cannot import this in the defaults init
+            limit_func = fullrange_limit_factory()
```

Comment:
Take this opportunity to change the default to percentile autolimits?

Reply:
done

Reply:
@tacaswell turns out this method (percentile_limit_factory) needs parameters, what should be the value?

Also I am not sure what is the preferred type for this, should it be a tuple of floats, an ndarray or a list?

Reply:
I would go with 5, 95